packages/components/src/slot-fill2/fill.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+import { createPortal } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useSlot } from './context';
+
+export default function Fill( { name, children } ) {
+	const slot = useSlot( name );
```

Comment:
This will trigger a re-render on this component when:
- a new slot has been added.
- a slot has been updated (`fillProps` or `ref`).
- a slot has been removed.